packages/eslint-plugin/rules/i18n-translator-comments.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Internal dependencies
+ */
+const {
+	TRANSLATION_FUNCTIONS,
+	REGEXP_PLACEHOLDER,
+	getTranslateFunctionName,
+	getTranslateFunctionArgs,
+	getTextContentFromNode,
+} = require( '../utils' );
+
+module.exports = {
+	meta: {
+		type: 'problem',
+		messages: {
+			missing:
+				'Translation function with placeholders is missing preceding translator comment',
+		},
+	},
+	create( context ) {
+		return {
+			CallExpression( node ) {
+				const {
+					callee,
+					loc: {
+						start: { line: currentLine },
+					},
+					parent,
+					arguments: args,
+				} = node;
+
+				const functionName = getTranslateFunctionName( callee );
+
+				if ( ! TRANSLATION_FUNCTIONS.has( functionName ) ) {
+					return;
+				}
+
+				const candidates = getTranslateFunctionArgs(
+					functionName,
+					args
+				)
+					.map( getTextContentFromNode )
+					.filter( Boolean );
+
+				if ( candidates.length === 0 ) {
+					return;
+				}
+
+				const hasPlaceholders = candidates.some( ( candidate ) =>
+					candidate.match( REGEXP_PLACEHOLDER )
```

Comment:
Minor: Either works just as well, but [`RegExp#test`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test)  would be the closest match for the intended behavior here, if we're not interested in the results of the match.

```suggestion
					REGEXP_PLACEHOLDER.test( candidate )
```

Reply:
This introduced a bug that took me a while to figure out. All fixed now: 6153febbbe0c0362df655cadfed98614a4fba0db

Reply:
That's interesting. I can't say that I was very familiar with that nuance. I think depending on how much we consider this "performance-critical"† code, I may be more inclined to just be fine with `String#match` if it means not needing to deal with errors which could come with not accommodating this.

† The [referenced MDN article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) does mention "slower execution" and it is indeed [somewhat slower](https://jsperf.com/test-vs-match-regex), though not egregiously so.